driver-core/src/test/java/com/datastax/driver/core/SSLTestBase.java

```diff
@@ -74,35 +76,61 @@ protected void connectWithSSLOptions(SSLOptions sslOptions) throws Exception {
+     * @param trustingServer whether the client should trust the server's certificate
+     * @return {@link com.datastax.driver.core.SSLOptions} with the given configuration for
      * server certificate validation and client certificate authentication.
      */
-    public SSLOptions getSSLOptions(Optional<String> keyStorePath, Optional<String> trustStorePath) throws Exception {
+    public SSLOptions getSSLOptions(SslImplementation sslImplementation, boolean clientAuth, boolean trustingServer) throws Exception {
 
         TrustManagerFactory tmf = null;
-        if(trustStorePath.isPresent()) {
+        if (trustingServer) {
             KeyStore ks = KeyStore.getInstance("JKS");
-            ks.load(this.getClass().getResourceAsStream(trustStorePath.get()), DEFAULT_CLIENT_TRUSTSTORE_PASSWORD.toCharArray());
+            ks.load(
+                this.getClass().getResourceAsStream(CCMBridge.DEFAULT_CLIENT_TRUSTSTORE_PATH),
+                CCMBridge.DEFAULT_CLIENT_TRUSTSTORE_PASSWORD.toCharArray());
 
             tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
             tmf.init(ks);
         }
 
-        KeyManagerFactory kmf = null;
-        if(keyStorePath.isPresent()) {
-            KeyStore ks = KeyStore.getInstance("JKS");
-            ks.load(this.getClass().getResourceAsStream(keyStorePath.get()), DEFAULT_CLIENT_KEYSTORE_PASSWORD.toCharArray());
 
-            kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
-            kmf.init(ks, DEFAULT_CLIENT_KEYSTORE_PASSWORD.toCharArray());
-        }
+        switch (sslImplementation) {
+            case JDK:
+                KeyManagerFactory kmf = null;
+                if (clientAuth) {
+                    KeyStore ks = KeyStore.getInstance("JKS");
+                    ks.load(
+                        this.getClass().getResourceAsStream(CCMBridge.DEFAULT_CLIENT_KEYSTORE_PATH),
+                        CCMBridge.DEFAULT_CLIENT_KEYSTORE_PASSWORD.toCharArray());
 
-        SSLContext sslContext = SSLContext.getInstance("TLS");
-        sslContext.init(kmf != null ? kmf.getKeyManagers() : null, tmf != null ? tmf.getTrustManagers() : null, new SecureRandom());
+                    kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+                    kmf.init(ks, CCMBridge.DEFAULT_CLIENT_KEYSTORE_PASSWORD.toCharArray());
+                }
 
-        return new SSLOptions(sslContext, SSLOptions.DEFAULT_SSL_CIPHER_SUITES);
+                SSLContext sslContext = SSLContext.getInstance("TLS");
+                sslContext.init(kmf != null ? kmf.getKeyManagers() : null, tmf != null ? tmf.getTrustManagers() : null, new SecureRandom());
+
+                return JdkSSLOptions.builder().withSSLContext(sslContext).build();
+
+            case NETTY_OPENSSL:
+                SslContextBuilder builder = SslContextBuilder
+                    .forClient()
+                    .sslProvider(OPENSSL)
```

Comment:
Netty throws an `Error` if it's not available ([source](https://github.com/netty/netty/blob/master/handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java#L208)), although I've not tested this.